### DISCUSSION TITLE
Request: specific alias to make build fail

### DISCUSSION THREAD

**Discussion by ymartin59** [cake-build/cake#3118] [Ideas]
I propose to create an alias which allows to make build fail without an explicit `throw new exception` after a `if` condition.
For instance `Fail(bool conditionToFail, string exceptionMessage)`
Cake may throw a specific exception in such case and report current task name and line location in cake build file.

**Comment by devlead** [comment]
This is a workaround, and not exactly what you're looking for, but still a tip of what you can do today.

What I've usually used in integration tests to fail fast is the [Xunit.Assert](https://www.nuget.org/packages/xunit.assert/) package.
One thing it has in handy is that it has sensible errors when comparing expected vs actual values. And `True` /  `False` has overloads that let you specify a user message.

An example script of that could look something like this:

## xunit.cake

```cake
#addin "nuget:?package=xunit.assert&version=2.4.1"
using static Xunit.Assert;

var target = Argument("target", "Default");

Task("MaybeFail")
    .Does(()=> Equal("Default", target));

Task("MaybeFail2")
    .Does(()=> True(HasArgument("target"), "Target is expected to be specified."));


Task("Default")
    .IsDependentOn("MaybeFail")
    .IsDependentOn("MaybeFail2");

RunTarget(target);
```

### Given Cake launched with `dotnet cake xunit.cake`

The expected result would be
```diff
    ========================================
    MaybeFail
    ========================================
    
    ========================================
    MaybeFail2
    ========================================
-   An error occurred when executing task 'MaybeFail2'.
-   Error: One or more errors occurred. (Target is expected to be specified.
-   Expected: True
-   Actual:   False)
-           Target is expected to be specified.
-           Assert.True() Failure
-   Expected: True
-   Actual:   False
```

### Given Cake launched with `dotnet cake .\xunit.cake --target=MaybeFail`

The expected result would be

```diff
    ========================================
    MaybeFail
    ========================================
-   An error occurred when executing task 'MaybeFail'.
-   Error: One or more errors occurred. (Assert.Equal() Failure
-             ↓ (pos 0)
-   Expected: Default
-   Actual:   MaybeFail
-             ↑ (pos 0))
-           Assert.Equal() Failure
-             ↓ (pos 0)
-   Expected: Default
-   Actual:   MaybeFail
-             ↑ (pos 0)
```

### Given cake launched with  `dotnet cake .\xunit.cake --target=MaybeFail2`

```diff
    ========================================
    MaybeFail2
    ========================================
    
+   Task                          Duration
+   --------------------------------------------------
+   MaybeFail2                    00:00:00.0119659
+   --------------------------------------------------
+   Total:                        00:00:00.0119659
```
